### PR TITLE
[Torch] TensorRT benchmark

### PR DESCRIPTION
**Pull request by fortianyou**
The pull request adds the following:
1. Tutorials and readme to the benchmark
2. List some test results
3. Update warmup and run_iterations in torch-tensorrt/perf_run.py

### CODE REVIEW

File: pytorch_blade/benchmark/detectron2/test_d2_benchmark.sh [hunk 1]
```diff
@@ -21,4 +22,4 @@ python3 run_blade.py --model testMaskRCNNFPN_pproc
 python3 run_blade.py --model testRetinaNet
 python3 run_blade.py --model testRetinaNet_scripted
 
-cd ..
+cd $oldpwd
```
Comment (Orion34-lanbo) [inline pytorch_blade/benchmark/detectron2/test_d2_benchmark.sh:25]: Nit: `cd -` will do the trick

File: pytorch_blade/benchmark/run_benchmark.sh [hunk 2]
```diff
@@ -45,3 +51,5 @@ bash benchmark/torch-tensorrt/test_trt_benchmark.sh 2>&1 | tee test_trt.log
 
 grep "|" test_d2.log
 grep "|" test_trt.log
+
+cd $oldpwd
```
Comment (Orion34-lanbo) [inline pytorch_blade/benchmark/run_benchmark.sh:55]: ditto

File: pytorch_blade/benchmark/torch-tensorrt/README.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+- Torch
+- Torch-TensorRT
+- TensorRT
+- Torch-Blade
+
+Note: Please note that for ONNX models, the user can convert the ONNX model to
+TensorRT serialized engine and then use this package.
+
+## Prerequisite
```
Comment (Orion34-lanbo) [inline pytorch_blade/benchmark/torch-tensorrt/README.md:17]: Nit: `Prerequisites` ?

File: pytorch_blade/benchmark/torch-tensorrt/perf_run.py [hunk 4]
```diff
@@ -73,7 +73,7 @@ def run_torch(model, input_tensors, params, precision):
     if precision == "fp16" or precision == "half":
         model = model.half()
 
-    iters = params.get("iterations", 20)
+    iters = params.get("iterations", 100)
```
Comment (Orion34-lanbo) [inline pytorch_blade/benchmark/torch-tensorrt/perf_run.py:76]: make this 100 as `BENCHMARK_ITER`?

File: pytorch_blade/benchmark/detectron2/test_d2_benchmark.sh (hunk 1 above)
Comment (fortianyou) [reply pytorch_blade/benchmark/detectron2/test_d2_benchmark.sh:25]: `cd -` only step back to the directory used before. But, here we would like to back a directory that executes this script.

Comment (Orion34-lanbo) [reply pytorch_blade/benchmark/detectron2/test_d2_benchmark.sh:25]: I see the entire script, if we only cd into a dir once, `cd -` is enough.

File: pytorch_blade/benchmark/torch-tensorrt/test_trt_benchmark.sh [hunk 5]
```diff
@@ -21,4 +21,4 @@ python3 perf_run.py --config=config/vgg16.yml
 python3 perf_run.py --config=config/yolov5.yml
 python3 perf_run.py --config=config/crnn.yml
 
-cd ..
+cd $oldpwd
```
Comment (qiuxiafei) [inline pytorch_blade/benchmark/torch-tensorrt/test_trt_benchmark.sh:24]: Seems useless ?

File: pytorch_blade/benchmark/torch-tensorrt/perf_run.py (hunk 4 above)
Comment (qiuxiafei) [reply pytorch_blade/benchmark/torch-tensorrt/perf_run.py:76]: +1

File: pytorch_blade/benchmark/torch-tensorrt/README.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+| Name | Supported Values | Description |
+| --- | --- | --- |
+| backend | all, torch, torch_tensorrt, tensorrt | Supported backends for inference. |
+| input | - | Input binding names. Expected to list shapes of each input bindings |
+| model | - | Configure the model filename and name |
+| filename | - | Model file name to load from disk. |
+| name | - | Model name |
+| runtime | - | Runtime configurations |
+| device | 0 | Target device ID to run inference. Range depends on available GPUs |
```
Comment (qiuxiafei) [inline pytorch_blade/benchmark/torch-tensorrt/README.md:69]: so, `1` is not a supported device value?

File: pytorch_blade/benchmark/torch-tensorrt/README.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+| vgg16 | Torch-Blade    | fp32        |       137.365 |     137.068 |           0.00727989 |         0.00729606 | 0.00744398 | 5.56986e-05 |
+| vgg16 | Torch-TensorRT | fp32        |       140.318 |     140.042 |           0.00712666 |         0.00716695 | 0.0073506  | 0.000575365 |
+| vgg16 | TensorRT       | fp32        |       139.23  |     138.601 |           0.00718237 |         0.00721714 | 0.00762804 | 0.000127496 |
+| vgg16 | Torch          | fp16        |       246.146 |     246.007 |           0.00406262 |         0.00406507 | 0.00413304 | 2.49436e-05 |
+| vgg16 | Torch-Blade    | fp16        |       400.434 |     401.282 |           0.00249729 |         0.0024925  | 0.00253454 | 3.45862e-05 |
+| vgg16 | Torch-TensorRT | fp16        |       381.676 |     364.289 |           0.00262002 |         0.00280076 | 0.00422715 | 0.000472002 |
+| vgg16 | TensorRT       | fp16        |       384.943 |     390.188 |           0.00259779 |         0.00256511 | 0.00267093 | 7.56818e-05 |
+
+The benchmark on vgg16 shows that TorchBlade has similar performance to
+Torch-TensorRT and TensorRT.
```
Comment (qiuxiafei) [inline pytorch_blade/benchmark/torch-tensorrt/README.md:117]: why is there a new line?

Comment (fortianyou) [reply pytorch_blade/benchmark/torch-tensorrt/README.md:117]: The linter will warn the too-long length line.

File: pytorch_blade/benchmark/torch-tensorrt/README.md (hunk 6 above)
Comment (fortianyou) [reply pytorch_blade/benchmark/torch-tensorrt/README.md:69]: > "Range depends on available GPUs"

The readme is modified from Torch-TensorRT's.

File: pytorch_blade/benchmark/torch-tensorrt/config/yolov5.yml [hunk 8]
```diff
@@ -9,8 +9,8 @@ input:
     - 640
   num_inputs: 1
 model: 
-  filename: models/yolov5.jit.pt
-  name: yolov5
+  filename: models/yolov5s.jit.pt
```
[the remark this answers is not in the document]
Comment (fortianyou) [reply pytorch_blade/benchmark/torch-tensorrt/config/yolov5.yml:12]: This one is smaller.

File: pytorch_blade/benchmark/torch-tensorrt/README.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+## Comparison (@Nvidia T4)
+
+| Model | Backend        | precision   |   Median(FPS) |   Mean(FPS) |   Median-Latency(ms) |   Mean-Latency(ms) |     99th_p |     std_dev |
+|------:|:---------------|:------------|--------------:|------------:|---------------------:|-------------------:|-----------:|------------:|
+| vgg16 | Torch          | fp32        |       117.516 |     116.689 |           0.00850948 |         0.00857261 | 0.00896431 | 0.000158592 |
+| vgg16 | Torch-Blade    | fp32        |       137.365 |     137.068 |           0.00727989 |         0.00729606 | 0.00744398 | 5.56986e-05 |
+| vgg16 | Torch-TensorRT | fp32        |       140.318 |     140.042 |           0.00712666 |         0.00716695 | 0.0073506  | 0.000575365 |
+| vgg16 | TensorRT       | fp32        |       139.23  |     138.601 |           0.00718237 |         0.00721714 | 0.00762804 | 0.000127496 |
+| vgg16 | Torch          | fp16        |       246.146 |     246.007 |           0.00406262 |         0.00406507 | 0.00413304 | 2.49436e-05 |
+| vgg16 | Torch-Blade    | fp16        |       400.434 |     401.282 |           0.00249729 |         0.0024925  | 0.00253454 | 3.45862e-05 |
```
Comment (minminsun) [inline pytorch_blade/benchmark/torch-tensorrt/README.md:112]: Torch-Blade is 4% faster than TensorRT, Great!

**Review by qiuxiafei — APPROVED** [review]

**Pull request merged**